src/frontend-pwa/src/services/app/useAppService.ts

```diff
@@ -38,9 +38,23 @@ const useAppService = () => {
     const setAppData = async (isOnline: boolean) => {
       if (isOnline) {
         try {
-          const data = await axios.get(`${constants.BACKEND_URL}/api/locations`);
-          saveDataToLocalStorage(constants.APP_DATA_KEY, data);
-          dispatch({ type: SET_APP_DATA, payload: data });
+          if (localStorageKeyExists(constants.UPDATE_ARRAY_KEY)) {
+            const data = getDataFromLocalStorage(constants.APP_DATA_KEY);
+            const updateData = await axios.post(`${constants.BACKEND_URL}/api/locations`, { updateArr: getDataFromLocalStorage(constants.UPDATE_ARRAY_KEY) });
+            Object.entries(updateData.data).forEach(([key, value]) => {
+              if (key in data) {
+                data[key] = value;
+              }
+            });
+            saveDataToLocalStorage(constants.UPDATE_ARRAY_KEY, data.data.updateArr);
```

Comment:
```suggestion
            saveDataToLocalStorage(constants.UPDATE_ARRAY_KEY, data.data.UPDATE_ARRAY_KEY);
```
Shouldn't these match since its 1:1

Reply:
@LocalNewsTV Just tried. Doesn't work that way